controllers/phjob_controller.go

```diff
@@ -145,9 +146,22 @@ func (r *PhJobReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		log.Info("found pod resource for PhJob")
 
 		phJob = phJob.DeepCopy()
-		phJob.Status.Phase = convertJobPhase(pod)
 		phJob.Status.StartTime = getStartTime(pod)
-		phJob.Status.FinishTime = getFinishTime(pod)
+
+		if phJob.Spec.Cancel == true && phJob.Status.Phase != primehubv1alpha1.JobCancelled {
+			if err := r.Client.Delete(ctx, pod); err != nil {
```

Comment:
Would it possible to delete with grace period = 0? Currently, we will wait 30sesc.

  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30

It should be in the 

```
kubectl explain pod.spec.terminationGracePeriodSeconds
```

Reply:
Done. Thx.

Reply:
LGTM